addons/packages/contour/1.20.0/bundle/config/values.yaml

```diff
@@ -0,0 +1,94 @@
+#@data/values
+#@overlay/match-child-defaults missing_ok=True
+
+---
+#! The underlying infrastructure provider. Options are aws, azure, docker and vsphere.
+#! This field is not required, but enables better validation and defaulting if provided.
+infrastructureProvider: ""
+
+#! The namespace in which to deploy Contour and Envoy.
+namespace: projectcontour
+
+#! Settings for the Contour component.
+contour:
+  #! The YAML contents of the Contour config file. See https://projectcontour.io/docs/v1.20.0/configuration/#configuration-file for more information.
+  configFileContents: {}
+
+  #! How many Contour pod replicas to have.
+  replicas: 2
+
+  #! Whether to enable PROXY protocol for all Envoy listeners.
+  useProxyProtocol: false
+
+  #! The Contour log level. Valid options are "info" and "debug".
+  logLevel: info
+
+#! Settings for the Envoy component.
+envoy:
+  #! Envoy service settings.
+  service:
+    #! The type of Kubernetes service to provision for Envoy. If not specified,
+    #! will default to "NodePort" for docker and vsphere and "LoadBalancer" for others.
+    type: ""
+
+    #! The desired load balancer IP.  If type is not "LoadBalancer", this field is ignored.
+    #! It is up to the cloud provider whether to honor this request.  If null, then load balancer IP
+    #! will be assigned by the cloud provider.
+    loadBalancerIP: null
```

Comment:
just curious, any reason this is null vs empty string?

Reply:
or i guess since `loadBalancerIP` was added first, why empty string vs null haha, seems like the same thing ultimately?

Reply:
Yeah i think ultimately either works as a "zero value", but let me double-check and make consistent.

Reply:
Double-checked this and they function equivalently; I switched all to use "" or 0 to mean "not specified" for consistency.